cosim/black-parrot-example/vivado/tcl/additional_constraints.tcl

```diff
@@ -0,0 +1,64 @@
+
+set bp_inst [join [get_cells -hier blackparrot]]
+
+set bp_core_period [get_property PERIOD [get_clocks -of_object [get_pins $bp_inst/aclk]]]
+set bp_rt_period [get_property PERIOD [get_clocks -of_object [get_pins $bp_inst/rt_clk]]]
+set bp_min_period [expr $bp_rt_period < $bp_core_period ? $bp_rt_period : $bp_core_period]
+
+set clint_inst [join [get_cells -hier -filter {(ORIG_REF_NAME == bp_me_clint_slice || REF_NAME == bp_me_clint_slice)}]]
+if {[llength clint_inst] != 1} {
+  error "More than one bp_me_clint_slice instance has been found"
```

Comment:
Why is this an error? For a multicore design this is reasonable

Reply:
Yeah my bad. I was only thinking about single core. I will change it to maybe a foreach statement.

Reply:
I'd maybe make it a function call and call in a loop